infra/gravity/node_commands.go

```diff
@@ -587,6 +604,66 @@ func (g *gravity) RunInPlanet(ctx context.Context, cmd string, args ...string) (
 	return out, nil
 }
 
+// IsLeader returns true if node is leader
+func (g *gravity) IsLeader(ctx context.Context) bool {
+	status, err := g.Status(ctx)
+	if err != nil {
+		return false
+	}
+	etcdLeaderKey := fmt.Sprintf("/planet/cluster/%s/master", status.Cluster.Cluster)
+
+	// TODO: use planet cmd once planet cli is updated s.t. `leader view` does
+	// not require args.
+	// leaderIP, err := g.RunInPlanet(ctx, "planet", "leader", "view")
+	leaderIP, err := g.RunInPlanet(ctx, "etcdctl", "get", etcdLeaderKey)
+	if err == nil && leaderIP == g.Node().PrivateAddr() {
+		return true
+	}
+	return false
+}
+
+// PartitionNetwork creates a network partition between this gravity node and
+// the cluster.
+func (g *gravity) ParitionNetwork(ctx context.Context) error {
+	status, err := g.Status(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	for _, node := range status.Cluster.Nodes {
+		cmdDropInput := fmt.Sprintf("iptables -I INPUT -s %s -j DROP", node.Addr)
+		if err := sshutils.Run(ctx, g.Client(), g.Logger(), cmdDropInput, nil); err != nil {
+			return trace.Wrap(err, cmdDropInput)
+		}
+		cmdDropOutput := fmt.Sprintf("iptables -I OUTPUT -s %s -j DROP", node.Addr)
+		if err := sshutils.Run(ctx, g.Client(), g.Logger(), cmdDropOutput, nil); err != nil {
+			return trace.Wrap(err, cmdDropOutput)
+		}
+	}
+	return nil
+}
+
+// UnpartitionNetwork removes network partition between this gravity node and
+// the cluster. If a network partition does not already exist, this will have
+// no effect.
+func (g *gravity) UnpartitionNetwork(ctx context.Context) error {
+	status, err := g.Status(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	for _, node := range status.Cluster.Nodes {
+		// TODO: look into using existing libs for manipulating network traffic.
+		cmdAcceptInput := fmt.Sprintf("iptables -I INPUT -s %s -j ACCEPT", node.Addr)
```

Comment:
I'm curious why you chose to insert another ACCEPT rule rather than deleting the DROP rule inserted by the partition method? Isn't the DROP one gonna be hit first?

Reply:
Oh, that's probably a better way to handle it.